### ISSUE TITLE
Unable to skip DNS checks with `flyctl deploy --dns-checks=false`

### ISSUE THREAD

**Issue by hilja** [superfly/flyctl#3318]
**Describe the bug**

If I deploy with a command:

```sh
flyctl deploy --remote-only --dns-checks=false --config=./foo/fly.toml -a=foo
```

I always get `Checking DNS configuration for foo.fly.dev` at the end of the deployment and it times out:

```
Error: read udp 10.145.233.127:56438->9.9.9.9:53: i/o timeout
```

My site works fine though and `dig` shows that the AAAA resolves. The timeout is a minute (or something) so my CI takes that much longer to complete.

Same error with `flyctl doctor -a=foo`.

There might be two issues:

1. Can't disable the DNS check
2. The DNS check doesn't work properly with Cloudflare DNS

I'm just tweaking the DNS, maybe I can get it working, but it would still be nice to disable that check.

* Operating system: Apple M2 Pro, Sonoma
* `fly version`: fly v0.2.9 darwin/arm64 Commit: 3fbcdb941df8e5e46a3d11a85404004b86e91dd3-dirty BuildDate: 2024-02-19T16:54:19Z

** Paste your `fly.toml`

```toml
app = "psl"
primary_region = "ams"
kill_signal = "SIGINT"
kill_timeout = 5
processes = []
swap_size_mb = 512

[env]
  PORT = "8080"
  METRICS_PORT = "8081"

[deploy]
  release_command = "bash ./psl/scripts/migrate.sh"

[build]
  dockerfile = "./scripts/Dockerfile"
  ignorefile = "./scripts/Dockerfile.dockerignore"

[[services]]
  internal_port = 8080
  processes = [ "app" ]
  protocol = "tcp"

    [services.concurrency]
      hard_limit = 25
      soft_limit = 20
      type = "connections"

    [[services.ports]]
      handlers = [ "http" ]
      port = 80
      force_https = true

    [[services.ports]]
      handlers = [ "tls", "http" ]
      port = 443

    [[services.tcp_checks]]
      grace_period = "1s"
      interval = "15s"
      restart_limit = 0
      timeout = "2s"

    [[services.http_checks]]
      interval = "30s"
      grace_period = "5s"
      method = "get"
      path = "/healthcheck"
      protocol = "http"
      timeout = "2s"
      tls_skip_verify = false
      [services.http_checks.headers]
```

** Command output: **

```text
Checking DNS configuration for psl.fly.dev
Error: read udp 10.145.233.127:56438->9.9.9.9:53: i/o timeout
```

**Comment by hilja** [comment]
Wait, is this the same check as "Checking DNS configuration for.."?

```
[[services.tcp_checks]]
  grace_period = "1s"
  interval = "15s"
  restart_limit = 0
  timeout = "2s"
```

**Comment by dangra** [comment]
Fixed by #3324 . thanks @hilja for the bug report.
